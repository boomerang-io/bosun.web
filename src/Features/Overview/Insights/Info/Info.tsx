import React from "react";
import { Tile } from "@boomerang-io/carbon-addons-boomerang-react";
import { Info as InfoProps } from "Types";
import styles from "./info.module.scss";

type Props = {
  info: InfoProps;
};

export function Info({ info }: Props) {
  return (
    <Tile
      style={{
        display: "flex",
        width: "100%",
        height: "10rem",
        alignItems: "center",
        padding: "0",
        marginBottom: "1rem",
        marginLeft: info.type === "gates" ? "0" : "1rem",
      }}
      data-testid="tile-info"
    >
      <div className={styles.counter} style={{ color: info.type === "gates" ? "#047cc0" : "#ffaa9d" }}>
        {info.count}
      </div>
      <div className={styles.divider} />
      <div className={styles.text}>
        <h3 className={styles.title}>{info.title}</h3>
        <p className={styles.content}>{info.content}</p>
      </div>
    </Tile>
  );
}

export default Info;
