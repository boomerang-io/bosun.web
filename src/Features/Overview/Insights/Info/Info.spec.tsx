import React from "react";
import Info from ".";
const info = {
  type: "test",
  title: "Test title",
  content: "Test content text",
  count: 0,
};
describe("Info --- Snapshot", () => {
  it("Capturing Snapshot of Info", async () => {
    const { baseElement } = (global as any).rtlRender(<Info info={info} />);
    expect(baseElement).toMatchSnapshot();
  });
});
