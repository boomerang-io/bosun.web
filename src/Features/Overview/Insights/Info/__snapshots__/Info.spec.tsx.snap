// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info --- Snapshot Capturing Snapshot of Info 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="bx--tile"
      data-testid="tile-info"
      style="display: flex; width: 100%; height: 10rem; align-items: center; padding: 0px; margin-bottom: 1rem; margin-left: 1rem;"
    >
      <div
        class="counter"
        style="color: rgb(255, 170, 157);"
      >
        0
      </div>
      <div
        class="divider"
      />
      <div
        class="text"
      >
        <h3
          class="title"
        >
          Test title
        </h3>
        <p
          class="content"
        >
          Test content text
        </p>
      </div>
    </div>
  </div>
</body>
`;
